test/test_ops.py

```diff
@@ -15,48 +14,12 @@
 from typing import Tuple
 
 
-class OpTester(object):
-    @classmethod
-    def setUpClass(cls):
-        cls.dtype = torch.float64
+class RoIOpTester:
+    dtype = torch.float64
 
-    def test_forward_cpu_contiguous(self):
-        self._test_forward(device=torch.device('cpu'), contiguous=True)
-
-    def test_forward_cpu_non_contiguous(self):
-        self._test_forward(device=torch.device('cpu'), contiguous=False)
-
-    def test_backward_cpu_contiguous(self):
-        self._test_backward(device=torch.device('cpu'), contiguous=True)
-
-    def test_backward_cpu_non_contiguous(self):
-        self._test_backward(device=torch.device('cpu'), contiguous=False)
-
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA unavailable")
-    def test_forward_cuda_contiguous(self):
-        self._test_forward(device=torch.device('cuda'), contiguous=True)
-
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA unavailable")
-    def test_forward_cuda_non_contiguous(self):
-        self._test_forward(device=torch.device('cuda'), contiguous=False)
-
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA unavailable")
-    def test_backward_cuda_contiguous(self):
-        self._test_backward(device=torch.device('cuda'), contiguous=True)
-
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA unavailable")
-    def test_backward_cuda_non_contiguous(self):
-        self._test_backward(device=torch.device('cuda'), contiguous=False)
-
-    def _test_forward(self, device, contiguous):
-        pass
-
-    def _test_backward(self, device, contiguous):
-        pass
-
-
-class RoIOpTester(OpTester):
-    def _test_forward(self, device, contiguous, x_dtype=None, rois_dtype=None, **kwargs):
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('contiguous', (True, False))
+    def test_forward(self, device, contiguous, x_dtype=None, rois_dtype=None, **kwargs):
```

Comment:
The previously manual cross-product between device and contiguous is now seamlessly handled by the parametrization, both for `test_forward` and `test_backward`